test(ast/estree): ESTree conformance tester include tests with `Infinity`

overlookmotel opened the pull request:
Now that we are comparing ASTs Oxc vs Acorn with a simple JSON string comparison (#9285), we can re-enable the tests for source files containing infinite numbers (e.g. `1e+9999`).

Our serializer outputs `f64::INFINITY` as `1e+400` in JSON, which `JSON.parse` interprets as JS value `Infinity`.

Bump `acorn-test262` to include commit https://github.com/oxc-project/acorn-test262/commit/ed8b455fd9775089444d53c09ea18fedf220da8b, which serializes `Infinity` in the same way.

overlookmotel commented:
> [!WARNING]
> <b>This pull request is not mergeable via GitHub because a downstack PR is open. Once all requirements are satisfied, merge this PR as a stack <a href="https://app.graphite.dev/github/pr/oxc-project/oxc/9286?utm_source=stack-comment-downstack-mergeability-warning" >on Graphite</a>.</b>
> <a href="https://graphite.dev/docs/merge-pull-requests">Learn more</a>

* **#9286** <a href="https://app.graphite.dev/github/pr/oxc-project/oxc/9286?utm_source=stack-comment-icon" target="_blank"><img src="https://static.graphite.dev/graphite-32x32-black.png" alt="Graphite" width="10px" height="10px"/></a> 👈 <a href="https://app.graphite.dev/github/pr/oxc-project/oxc/9286?utm_source=stack-comment-view-in-graphite" target="_blank">(View in Graphite)</a>
* **#9285** <a href="https://app.graphite.dev/github/pr/oxc-project/oxc/9285?utm_source=stack-comment-icon" target="_blank"><img src="https://static.graphite.dev/graphite-32x32-black.png" alt="Graphite" width="10px" height="10px"/></a>
* **#9284** <a href="https://app.graphite.dev/github/pr/oxc-project/oxc/9284?utm_source=stack-comment-icon" target="_blank"><img src="https://static.graphite.dev/graphite-32x32-black.png" alt="Graphite" width="10px" height="10px"/></a>
* **#9283** <a href="https://app.graphite.dev/github/pr/oxc-project/oxc/9283?utm_source=stack-comment-icon" target="_blank"><img src="https://static.graphite.dev/graphite-32x32-black.png" alt="Graphite" width="10px" height="10px"/></a>
* **#9282** <a href="https://app.graphite.dev/github/pr/oxc-project/oxc/9282?utm_source=stack-comment-icon" target="_blank"><img src="https://static.graphite.dev/graphite-32x32-black.png" alt="Graphite" width="10px" height="10px"/></a>
* `main`
<hr />

## How to use the Graphite Merge Queue

Add either label to this PR to merge it via the merge queue:
- <kbd>0-merge</kbd> - adds this PR to the back of the merge queue
- <kbd>hotfix</kbd> - for urgent hot fixes, skip the queue and merge this PR next

You must have a Graphite account in order to use the merge queue. Sign up using [this link](https://app.graphite.dev/invite/github/oxc-project?ref=merge-queue-instructions-in-stack-comment&prId=9286).

<h2></h2>
<sub>

An organization admin has enabled the [Graphite Merge Queue](https://graphite.dev/docs/graphite-merge-queue) in this repository.

</sub>

<sub>

Please do not merge from GitHub as this will restart CI on PRs being processed by the merge queue.

</sub>

<h2></h2>

This stack of pull requests is managed by <a href="https://graphite.dev?utm-source=stack-comment"><b>Graphite</b></a>. Learn more about <a href="https://stacking.dev/?utm_source=stack-comment">stacking</a>.
<!-- Current dependencies on/for this PR: -->

Boshen commented:
### Merge activity

* **Feb 21, 9:21 PM EST**: The merge label '0-merge' was detected. This PR will be added to the [Graphite merge queue](https://app.graphite.dev/merges?org=oxc-project&repo=oxc) once it meets the requirements.

overlookmotel committed:
test(ast/estree): ESTree conformance tester include tests with `Infinity` (#9286)

Now that we are comparing ASTs Oxc vs Acorn with a simple JSON string comparison (#9285), we can re-enable the tests for source files containing infinite numbers (e.g. `1e+9999`).

Our serializer outputs `f64::INFINITY` as `1e+400` in JSON, which `JSON.parse` interprets as JS value `Infinity`.

Bump `acorn-test262` to include commit https://github.com/oxc-project/acorn-test262/commit/ed8b455fd9775089444d53c09ea18fedf220da8b, which serializes `Infinity` in the same way.
